### PR TITLE
get secrets from vault and verify incoming requests from Github

### PR DESCRIPTION
**Pull request by hillalex**
* Added logic to verify request are coming from GitHub 
* Borrowing pattern (simplified for this simpler use case) from orderly-web-deploy for getting secrets out of the vault.
* Image now being built on TC

Next steps: deploy this on support and proxy it through the main Montagu deploy

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -1,7 +1,9 @@
 {
   "youtrack_instance_name": "vimc",
-  "youtrack_token": "<PERM_TOKEN>",
+  "youtrack_token": "VAULT:secret/webhook/youtrack:token",
```
Comment (richfitz) [inline config.json]: I wonder (and this is really a special case of https://vimc.myjetbrains.com/youtrack/issue/VIMC-2564) is to consider moving the secret keys from `secret/webhook/...` to `secret/youtrack-integration/...` to be slightly more informative.

File: save_settings [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            updated, v = resolve_secret(v)
+            if updated:
+                dict[k] = v
+
+
+def save_settings():
+    with open("config.json", 'r') as f:
+        data = json.load(f)
+        resolve_secrets(data)
+    with open("config.final.json", 'a'):  # Create file if does not exist
```
Comment (richfitz) [inline save_settings]: you can omit 45-46 if you change 47 to open the file with `w+`

**Review by richfitz — CHANGES_REQUESTED** [review]
This looks good to me, though it's going to be difficult to test without just using it I imagine! Two tiny things to look at

File: README.md [hunk 3]
```diff
@@ -21,14 +21,25 @@ Actions taken by the webhook:
...
+1. a secret token used by GitHub to secure the webhook: see https://developer.github.com/webhooks/securing/
 1. a dictionary from GitHub logins to YouTrack usernames
 
-See the `sample.config.json` file.
+If a config value is provided in the format "VAULT:path:key" in will be resolved by looking up the secret from vault.
+
+## Deployment
+The app runs inside a docker container and is mapped to port 4567 on the host machine.
+To deploy:
+1. `pip3 intall -r requirements.txt`
```
Comment (EmmaLRussell) [inline README.md]: Typo 'intall'

File: src/app/main.py [hunk 4]
```diff
@@ -1,50 +1,46 @@
...
 
 from .YouTrackHelper import YouTrackHelper
 
-app = Flask(__name__)
-
 
 def load_settings():
     with open("config.json", 'r') as f:
         data = json.load(f)
+    with open("config.json", 'w') as f:
```
Comment (EmmaLRussell) [inline src/app/main.py:13]: I'm not sure why this is here. Maybe a comment?

**Review by EmmaLRussell — CHANGES_REQUESTED** [review]

File: save_settings (hunk 2 above)
Comment (hillalex) [reply save_settings]: Oh cool, thanks

File: config.json (hunk 1 above)
Comment (hillalex) [reply config.json]: Good idea

**Review by richfitz — APPROVED** [review]

**Review by EmmaLRussell — APPROVED** [review]

**Pull request merged**